### PR TITLE
website: Refactor task page routes and repetition

### PR DESCRIPTION
**Pull request by rjmacarthy**
Hey,

Through careful refactoring, we were able to make the code more DRY (Don't Repeat Yourself) by creating a single, reusable component that dynamically renders the correct task based on the type passed to it. This not only streamlines the codebase but also ensures consistency across all routes. Additionally, we have fixed an error that previously caused the app to show a disabled button instead of a "no-tasks" interface #924. These improvements will lead to a smoother user experience and make the code easier to maintain.

If you have any questions or concerns please let me know.

### CODE REVIEW

**Comment by andreaskoepf** [comment]
@rjmacarthy did you talk to @fozziethebeat on discord about this or was it discussed in an issue here on gh? In general it is better to first clarify with the code owner. I am no member of the web-team but if you make far reaching changes without prior discussion you risk that your work is lost because it never gets merged.

**Comment by rjmacarthy** [comment]
> @rjmacarthy did you talk to @fozziethebeat on discord about this or was it discussed in an issue here on gh? In general it is better to first clarify with the code owner. I am no member of the web-team but if you make far reaching changes without prior discussion you risk that your work is lost because it never gets merged.

Hey @andreaskoepf,

Thank you for bringing this to my attention. Apologies, I will make sure to reach out to @fozziethebeat and discuss any proposed changes with them before proceeding. I will also ensure that any discussions take place in the appropriate forum, such as a Discord channel or GitHub issue, to ensure that the changes align with the project's goals. My apologies for not following this process in the first place. It should be mentioned that the issue that was raised is now fixed and I hope that this can be taken into consideration when reviewing the changes. If you would prefer that I close this pull request in favour of the original code then please let me know.

Thank you!

File: website/src/pages/label/label_prompter_reply.tsx
Comment (fozziethebeat) [inline website/src/pages/label/label_prompter_reply.tsx:23]: Where did these title and description texts go? I don't see locale files with these strings.

**Review by fozziethebeat — CHANGES_REQUESTED** [review]
I'm generally fine with refactoring like this.  But like Andreas said, its good to announce and discuss this in Discord.  This kind of big sweeping refactoring could impact other PRs that people have going on a make their merges hard.

Before I approve please address the one comment I made and announce in the #website channel this change to see if it will make anyone else's work harder.

File: website/src/pages/label/label_prompter_reply.tsx
Comment (rjmacarthy) [reply website/src/pages/label/label_prompter_reply.tsx:23]: Thank you @fozziethebeat, I understand and appreciate where you're both coming from.  The reason this change also includes a refactor was to prevent from adding error handing code for no available tasks in ten different files, now we have it in just one.  In terms of the title and description we are able to leverage the already existing `label` and `desc` properties from the locale files to apply to the `meta` tags in the header.  I hope that helps you understand the change more clearly?  Any problems please let me know.

https://github.com/LAION-AI/Open-Assistant/blob/main/website/public/locales/en/tasks.json

File: website/src/components/TaskPage/TaskPage.tsx
Comment (AbdBarho) [inline website/src/components/TaskPage/TaskPage.tsx:19]: could you please rename this variable to `taskInfo`?

File: website/src/lib/constants.ts
Comment (AbdBarho) [inline website/src/lib/constants.ts:31]: could you maybe add types to this? something like:

```ts
const apiHooksByType: { [taskType in TaskType]: (...args: any[]) => any } = ...
```
or even something better defined if you can.

**Review by AbdBarho — COMMENTED** [review]
Thanks!

File: website/src/lib/constants.ts
Comment (rjmacarthy) [reply website/src/lib/constants.ts]: Yes, no problem.

File: website/src/components/TaskPage/TaskPage.tsx
Comment (rjmacarthy) [reply website/src/components/TaskPage/TaskPage.tsx]: Sure, done.

**Review by AbdBarho — APPROVED** [review]

**Review by fozziethebeat — APPROVED** [review]
Shipping it!

**Pull request merged**
